src/main/java/org/trigon/hardware/phoenix6/talonfx/TalonFXMotor.java

```diff
@@ -29,68 +47,146 @@ public TalonFXMotor(int id, String motorName, String canbus) {
         motorIO.optimizeBusUsage();
     }
 
+    /**
+     * Updates the motor and logs the inputs.
+     */
     public void update() {
         motorIO.updateMotor();
         Logger.processInputs("Motors/" + motorName, motorInputs);
     }
 
+    /**
+     * Gets the ID of the motor.
+     *
+     * @return the ID of the motor
+     */
     public int getID() {
         return id;
     }
 
+    /**
+     * Sets the physics simulation of the motor. Needed for the motor to be used in simulation with accurate physics.
+     *
+     * @param physicsSimulation the simulation
+     */
     public void setPhysicsSimulation(MotorPhysicsSimulation physicsSimulation) {
         motorIO.setPhysicsSimulation(physicsSimulation);
     }
 
+    /**
+     * Applies both the real and simulation configurations to the motor.
+     *
+     * @param realConfiguration       configuration to be used in real life
+     * @param simulationConfiguration configuration to be used in simulation
+     */
     public void applyConfigurations(TalonFXConfiguration realConfiguration, TalonFXConfiguration simulationConfiguration) {
         if (RobotHardwareStats.isSimulation())
             motorIO.applyConfiguration(simulationConfiguration);
         else
             motorIO.applyConfiguration(realConfiguration);
     }
 
+    /**
+     * Applies the configuration to be used both in real life and in simulation.
+     *
+     * @param simulationAndRealConfiguration the configuration
+     */
     public void applyConfiguration(TalonFXConfiguration simulationAndRealConfiguration) {
         motorIO.applyConfiguration(simulationAndRealConfiguration);
     }
 
+    /**
+     * Applies the configuration to be used when {@link RobotHardwareStats#isSimulation()} is false.
+     *
+     * @param realConfiguration the configuration
+     */
     public void applyRealConfiguration(TalonFXConfiguration realConfiguration) {
         if (!RobotHardwareStats.isSimulation())
             motorIO.applyConfiguration(realConfiguration);
     }
 
+    /**
+     * Applies the configuration to be used in simulation.
+     *
+     * @param simulationConfiguration the configuration
+     */
     public void applySimulationConfiguration(TalonFXConfiguration simulationConfiguration) {
         if (RobotHardwareStats.isSimulation())
             motorIO.applyConfiguration(simulationConfiguration);
     }
```

Comment:
keep a similar javadoc fpr each one and explain why you'd need different configs for sim and real